odoo/orm/models.py

```diff
@@ -247,23 +247,20 @@ def add_default(name, field):
                     setattr(self, name, field)
                     field.__set_name__(self, name)
 
-            add('id', Id(automatic=True))
-            add_default('display_name', Char(
-                string='Display Name', automatic=True,
-                compute='_compute_display_name',
-                search='_search_display_name',
-            ))
+            # make sure `id` field is still a `fields.Id`
+            if hasattr(self, 'id') and not isinstance(self.id, Id):
+                raise TypeError(f"id field {self.id} of {self} is not an instance of Id")
```

Comment:
`self.id` always exists.
```suggestion
            if not isinstance(self.id, Id):
                raise TypeError(f"Field {self.id} is not an instance of fields.Id")
```